once/Dockerfile

```diff
@@ -0,0 +1,150 @@
+
+## Allow the user to run sudo, and run it without password.
+RUN set -eux; \
+	usermod --append --groups sudo authgear; \
+	printf "authgear ALL=(ALL) NOPASSWD:ALL\n" > /etc/sudoers.d/900-authgear
+
+## Install PostgreSQL 16.6
+##
+##
+## We have to install the package postgresql-common first.
+## The package postgresql-common installs the following file
+##
+## /etc/postgresql-common/createcluster.conf
+##
+## In this file, there is an option create_main_cluster.
+## We have to uncomment that option, and set it to false, so that
+## when the package postgresql-MAJOR is installed, it does not automatically
+## run initdb to create a database that we are not going to use.
+##
+##
+## The installed files belong to the user "postgres".
+## We change them back to "authgear".
+## \! -path '/proc/*' to skip searching in /proc/ as that would result in file not found error.
+##
+##
+## The following files are installed sample configuration files.
+##
+## /usr/share/postgresql/16/pg_ident.conf.sample
+## /usr/share/postgresql/16/pg_service.conf.sample
+## /usr/share/postgresql/16/postgresql.conf.sample
+## /usr/share/postgresql/16/pg_hba.conf.sample
+##
+## In particular, we want to patch the following files:
+##
+## /usr/share/postgresql/16/postgresql.conf.sample
+##
+## By default, there is a line `#listen_addresses = 'localhost'`.
+## We want to uncomment it and change the value to '*', so that
+## the Docker host can access it.
+##
+## /usr/share/postgresql/16/pg_hba.conf.sample
+##
+## By default, pg_hba.conf does not specify how to authenticate connection NOT from the loopback address.
+## We want to add a line to specify it.
+## initdb will replace the token `@authmethodhost@` and `@authmethodlocal@` in this file with the given value of
+## --auth-host and --auth-local respectively.
+## See https://doxygen.postgresql.org/initdb_8c.html
+## So the line we add should use the token.
+ENV PG_MAJOR=16
+RUN set -eux; \
+	apt-get update; \
+	apt-get install -y --no-install-recommends --no-install-suggests \
+		postgresql-common; \
+	sed -E -i "s,^#(create_main_cluster)\\s*=\\s*true,\\1 = false," \
+		/usr/share/postgresql-common/createcluster.conf \
+		/etc/postgresql-common/createcluster.conf \
+	; \
+	apt-get install -y --no-install-recommends --no-install-suggests \
+		postgresql-16=16.6-\*; \
+	rm -rf /var/lib/apt/lists/*; \
+	find / \
+		-user postgres \
+		\! -path '/proc/*' \
+		-exec chown authgear:authgear '{}' \; ; \
+	sed -E -i "s,^#?(listen_addresses)\\s*=\\s*\\S+,\\1 = '*'," "/usr/share/postgresql/$PG_MAJOR/postgresql.conf.sample"; \
+	printf "host\tall\tall\tall\t%s\n" "@authmethodhost@" >> "/usr/share/postgresql/$PG_MAJOR/pg_hba.conf.sample"
+
+ENV PATH=/usr/lib/postgresql/$PG_MAJOR/bin:$PATH
+
+## Install Redis 7.0.x
+RUN set -eux; \
+	apt-get update; \
+	apt-get install -y --no-install-recommends --no-install-suggests \
+		redis=5:7.0.\*; \
+	rm -rf /var/lib/apt/lists/*
```

Comment:
Not sure how we are going to setup the build pipeline, but line above L139 will always be cached until we manually delete the cache so packages might not be up-to-date.

Reply:
The first line `from ubuntu:noble` will be invalidated by `docker build --pull`. So in the build pipeline, there is no cache.

Reply:
I see. If we build with --pull then there should be no problem.